homework/list.js

```diff
@@ -1,79 +1,136 @@
-    _listNoteContent = list;
-  }
-
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
-
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
+    NoteListManager.prototype = {
+    /**
+     * add a eventlistener to listen click event 
+     * and initial wrapper
+     * and call fetchList to fetech data,
+     * after fetch , draw the list
+     * 
+     * @this {NoteListManager}
+     */
+        start() {
+            window.addEventListener('click',this);
+            this._wrapper = document.querySelector('#note-list-wrapper');  
+            this.fetchList()
+            .then((function(data) {
+                this.updateList(data);
+                this.drawList();
+                this.preloadFirstNote();
+            }).bind(this));
+        },
+        /** 
+         * Handle click event.
+         * If users click on note-title, call onNoteOpen to handle.
+         * @this {NoteListManager}
+         * @param {Object} event - mouse click event
+         */
+        handleEvent(event){
+            switch(event.type){
+                case 'click':
+                    if (event.target.classList.contains('note-title') ) {
+                        this.onNoteOpen(event)
+                    }
+                    break;
+            } 
+        },
+        /**
+         * Dispatch a custom event when mouse click at the note-title.
+         * Also determine the content to be showed.
+         * @this {NoteListManager}
+         * @param {Object} event- mouse click event.
+         */
+        onNoteOpen(event) {
+            var id = event.target.dataset.noteId;
+            var content = this._listNoteContent[id];
+            window.dispatchEvent(new CustomEvent('note-open',
+                                                 { detail: content }));
+        },
+        /**
+         * Show the Note for start stage.
+         * @this {NoteListManager}
+         */
+        preloadFirstNote() {
+            if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by a conditional statement only with one branch, it's better to put the `if...else` in the caller, rather than the callee.